app/services/pagelets/pagelet.rb

```diff
@@ -0,0 +1,24 @@
+module Pagelets
+  class Pagelet
+    attr_reader :name, :partial, :priority, :opts
+
+    def initialize(name, partial, priority, opts)
+      @name = name
+      @partial = partial
+      @priority = priority
+      @opts = opts
+    end
+
+    def <=>(other)
+      priority <=> other.priority
+    end
+
+    def snake_name
+      opts[:id] || @name.gsub(/\s+/, "_").underscore
```

Comment:
Wait, I don't like this at all. We are mixing two things together, this should be generic and clear enough to be used in arbitrary snippets.

So I suggest to rename this method to `pagelet_id` or simply `id` and also prefix the default id:

```ruby
def id
  opts[:id] || 'pagelet-id-' + @name.gsub(/\s+/, "_").underscore
end
```